MINOR: Add support for bootstrap-server in create_topic

soondenana opened the pull request:
create_topic method uses --zookeeper option to create a new topic. This
change adds support for it to switch to --bootstrap-server config if it
is supported.

As "--zookeeper" option is deprecated and "--bootstrap-server" is
preferred, this change allows us to move away from the deprecated
option.

Tested by launching tests and confirming they pass and use the new
--bootstrap-server option.

*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

soondenana committed:
MINOR: Add support for bootstrap-server in create_topic

create_topic method uses --zookeeper option to create a new topic. This
change adds support for it to switch to --bootstrap-server config if it
is supported.

As "--zookeeper" option is deprecated and "--bootstrap-server" is
preferred, this change allows us to move away from the deprecated
option.

Tested by launching tests and confirming they pass and use the new
--bootstrap-server option.

soondenana commented:
cc @mumrah , @jsancio, @hachikuji

jsancio commented on tests/kafkatest/services/kafka/kafka.py:
```diff
@@ -746,6 +763,18 @@ def describe_consumer_group(self, group, node=None, command_config=None):
+        if node.version.supports_bootstrap_server():
+            connection_setting = "--bootstrap-server %s" % self.bootstrap_servers(self.security_protocol)
+        else:
+            connection_setting = "--zookeeper %s" % self.zk_connect_setting()
```
Or we can support `--bootstrap-server` only. Wouldn't `node.version.supports_bootstrap_server()` always return true?

soondenana replied:
@hachikuji  mentioned that these tests can run against a lower version of Kafka cluster, so it is possible for it to run against say 2.1 version.

jsancio replied:
Hmmm. If that is the case then at least `kafka-leader-election.sh` won't exist in older versions of Kafka. I think we added it in 2.3.

soondenana replied:
Talked to @hachikuji  and it seems that the test in discussion is client_compatibility_features_test.py. I looked at the test and I think you are right, we don't need older zookeeper connection. I will update the code as you suggested and post a new commit.

jsancio commented on tests/kafkatest/services/kafka/kafka.py:
```diff
@@ -487,6 +489,20 @@ def set_unclean_leader_election(self, topic, value=True, node=None):
 
+    def perform_leader_election(self, topic, partition, election_type, node=None):
+        """
+        Perform leader election for the partition of the topic passed in as argument.
```
Maybe this should read:
> Perform leader election for the partition of the topic passed in as argument and waits until leadership changes replica.

soondenana replied:
Done.

jsancio reviewed (approved):
LGTM. Left some minor comments.

soondenana committed:
Took care of Jose's review comment

soondenana committed:
Removed supports_bootstrap_server and always use bootstrap server

soondenana committed:
Update kakfa-topics.sh execution to use security configurations (#690)

The original commit of this PR updated code for `kafka-topics.sh' command to use the
new --bootstrap-server option. That exposed a few issues that this
commit fixes:

1. Some command line flags are not compatible with --bootstrap-server
2. SASL related broker deployements were not working.

For first issue, following switches are not compatible with
--bootstrap-server option:
1. --delete-config
2. --if-exists
3. --if-not-exists

We only use `--if-not-exists' in systest, so this change checks for this
option and switches to --zookeeper if this option is present.

For the second issue, this change now get the client security
configuration from the SecurityConfig object and passes that as
--command-conifg option. It also specifies the JASS and Kerberos
configuration as environment variables.

Tested by running replication_test to confirm that test cases pass for
both SASL enabled and plain Kafka brokers. Also it works when
`--if-not-exists` option is specified.

soondenana commented:
Closing this as code has diverged and some of it is already been taken care of as part of KIP-500.
